lib/nerdtree/creator.vim

```diff
@@ -38,6 +38,24 @@ function! s:Creator.BufNamePrefix()
     return 'NERD_tree_'
 endfunction
 
+" FUNCTION: s:Creator.CreateExplorerTree(dir) {{{1
+function! s:Creator.CreateExplorerTree(dir)
+    try
+        let path = g:NERDTreePath.New(a:dir)
+    catch /^NERDTree.InvalidArgumentsError/
+        call nerdtree#echo('Invalid directory name:' . a:dir)
+        return
+    endtry
+
+    let creator = s:Creator.New()
+    if getbufinfo('%')[0].changed
```

Comment:
Hi, Thanks for this quick implementation, I like this approach with a new function instead of touching the `createWindowTree` function itself. I have a few thoughts about this section, I appreciate what you have done here with the splitting, But doesn't it make more sense if we replace these lines:
```vim
    if getbufinfo('%')[0].changed
        let l:splitLocation = g:NERDTreeWinPos ==# 'left' || g:NERDTreeWinPos ==# 'top' ? 'topleft ' : 'botright '
        let l:splitDirection = g:NERDTreeWinPos ==# 'left' || g:NERDTreeWinPos ==# 'right' ? 'vertical' : ''
    endif
    silent! execute l:splitLocation . l:splitDirection . ' new'
```
with this:
```vim
    if !isdirectory(expand('%:p:r'))
        silent! execute 'enew'
    end
```
Since it makes the behavior more like the netrw Explore command across the board. (also on that note, what do you think about renaming it from `NERDTreeExplorer` to `NERDTreeExplore`?)

Reply:
Hi, the splitting to a new window only occurs when the current buffer is modified. This is to address the behavior on a modified buffer that you mentioned.

Reply:
Maybe I'm getting it wrong, but we do not discard changes if we just create a new buffer. the modified buffer will continue to exist afterward.
Explore does the same thing, It creates a new netrw buffer and just changes the buffer in the active window(instead of closing or saving it).

We can also introduce splitting behavior with a new command like `NERDTreeSplitExplore` but instead of limiting it to the modified buffers we can apply it to every situation for the sake of consistency; With this change, users can choose between either opening a tree in the current window or create a new split for it without random behavior when the state of file changes.

If you have any strong opposition against doing it this way just let me know so we can talk about it.

Reply:
On a modified buffer, we cannot create a new buffer using `enew`. From documentation, enew "fails when changes
have been made to the current buffer, unless 'hidden
' is set or 'autowriteall' is set and the file can be written." I would say that for most buffers, 'autowriteall' and 'hidden' are not set like in my gif below. After modifying a nerdtree source file, I use `Explore` and it opens a netrw buffer in a new split. This is because a hidden buffer is the only buffer that can have changes without being the active buffer.
![output](https://github.com/preservim/nerdtree/assets/20745877/7f1b4959-2c15-4f7d-85bd-da4de5ef47b1)

Because I want to avoid handling or modifying the 'hidden' and 'autowriteall' options or creating separate similar options for nerdtree, I split like `Explore` command on a modified buffer. Then on an unmodified buffer, I replace the current buffer with the nerdtree window tree in the same way as the netrw buffer replacing the current buffer. 

Regarding a `NERDTreeSplitExplore` command, I am open to creating that. However, I want to come to an agreement on what `NERDTreeExplore` should do first. Thanks!

Reply:
Hey, Sorry for my late response; At the moment I'm a little busy with my day job, and I had to do some testing before I can speak about this PR.
I wasn't aware of this behavior since I always had `set hidden` in my vimrc. Thanks for being so patient with me and informing me about these flags. Now that we are on the same page, wouldn't it be better if we actually check for these flags and act accordingly? After all, we both want to make this command behave similarly to the Explore command.
So here is what I came up with for this part, Give it a look and let me know if there is anything else we should be anticipating in order to keep these 2 commands more aligned.
```vim
    if getbufinfo('%')[0].changed && !&hidden && !&autowriteall
        let l:splitLocation = g:NERDTreeWinPos ==# 'left' || g:NERDTreeWinPos ==# 'top' ? 'topleft ' : 'botright '
        let l:splitDirection = g:NERDTreeWinPos ==# 'left' || g:NERDTreeWinPos ==# 'right' ? 'vertical' : ''
        silent! execute l:splitLocation . l:splitDirection . ' new'
    elseif !isdirectory(expand('%:p:r'))
        silent! execute 'enew'
    endif

```

Reply:
No worries on the late reply! I understand if you have to handle your day job first. It is your day job after all 😆. Thank you for your questions! They are forcing me to be more thorough and examine my own assumptions in my own explanations. I am working to be more thorough without being too pedantic.

Coming back to the code, I agree with the handling for `hidden` and `autowriteall` and I will add that in the next commit. For the  `!isdirectory(expand('%:p:r'))` conditional, why do you not want to `enew` on a di?

Reply:
No worries on the late reply! I understand if you are busy with your day job. It is your day job after all 😆. Thank you for your questions. They forced me to be more thorough and examine my own assumptions in my explanations. Please let me know though if I become too thorough and approach being pedantic. 

Back to the code, I agree with your handling of the `hidden` and `autowriteall` flags. I'll push a commit with all the changes we've discussed after one last discussion lol. For the `!isdirectory(expand('%:p:r'))` conditional, why do you not want to use `enew` on a directory buffer? If we are using nerdtree only, when would we be in a directory buffer to not call `enew`? This is for clarification only and I do not see any impact to keeping this aligned to `:Explore`. Thanks!

Reply:
I was under the impression that createWindowTree will set `bufhidden` to `wipe` and this way we can prevent the NERDTree buffer from appearing in the history when we run it multiple times on the same buffer; But I was wrong about that, Turns out that common buffer options sets the `bufhidden` to `hide` and it can still appear as the previously visited location.

In order to achieve this result I had to change it to this.
```vim
    if getbufinfo('%')[0].changed && !&hidden && !&autowriteall
        let l:splitLocation = g:NERDTreeWinPos ==# 'left' || g:NERDTreeWinPos ==# 'top' ? 'topleft ' : 'botright '
        let l:splitDirection = g:NERDTreeWinPos ==# 'left' || g:NERDTreeWinPos ==# 'right' ? 'vertical' : ''
        silent! execute l:splitLocation . l:splitDirection . ' new'
    else
        silent! execute 'enew'
    endif

    call creator.createWindowTree(a:dir)
    " Notice me!
    setlocal bufhidden=wipe

```

Reply:
This looks good to me!  I got these changes and renamed everything to use `Explore` instead of `Explorer` like we discussed earlier.